### ISSUE TITLE
Investigate: usage of async event handlers

### ISSUE THREAD

**Issue by stsrki** [Megabit/Blazorise#4113]
https://www.codeproject.com/Articles/5341837/Asynchronous-Events-in-Csharp

**Comment by David-Moreira** [comment]
This reminds me of an `Event`  helper StevenSanderson had made. Didn't that support async? 
Do you remember what I am talking about?

**Comment by stsrki** [comment]
Yes, I remember. I think Steve's solution was more for events that don't trigger re-render.

**Comment by David-Moreira** [comment]
Help me remind here, but don't you still need to trigger a `StateHasChanged` somewhere in the case of standard dotnet events? I believe they are not "tracked" in anyway?

**Comment by stsrki** [comment]
You right. `StateHasChanged` is needed for regular events.
